routing/router.go

```diff
@@ -1101,3 +1104,54 @@ func (r *ChannelRouter) AddProof(chanID lnwire.ShortChannelID,
 	info.AuthProof = proof
 	return r.cfg.Graph.UpdateChannelEdge(info)
 }
+
+// getGraphDiameter is a function that determines the maximum distance between
+// two nodes in the graph, where the distance is calculated as the minimum number
+// of hops from src to dest
+func (r *ChannelRouter) getGraphDiameter() (uint8,error) {
+	// local variable to represent the graph diameter
+	var diam uint8
+	// If Cache is non-empty, then return Cached value
+	if r.diamCache != 0 {
```

Comment:
I feel like since we compute the paths in diamPath anyways, we could just pop them into a maxHeap and then just make changes to the heap nodes whenever nodes are attached/detachex.

This seems like a good option, what are your thoughts?